pkg/ccl/backupccl/backupinfo/manifest_handling.go

```diff
@@ -529,6 +554,44 @@ func WriteBackupLock(
 	return cloud.WriteFile(ctx, defaultStore, lockFileName, bytes.NewReader([]byte("lock")))
 }
 
+// WriteSlimBackupManifestWithMetadataSSTs writes a "slim" version of manifest
+// to `exportStore`. This version currently has the alloc heavy `Files` repeated
+// field nil'ed out, and written to an accompanying SST instead.
+func WriteSlimBackupManifestWithMetadataSSTs(
+	ctx context.Context,
+	exportStore cloud.ExternalStorage,
+	encryption *jobspb.BackupEncryptionOptions,
+	kmsEnv cloud.KMSEnv,
+	manifest *backuppb.BackupManifest,
+) error {
+	if err := writeSlimBackupManifest(ctx, exportStore, backupbase.SlimBackupManifestName, encryption,
+		kmsEnv, manifest); err != nil {
+		return errors.Wrap(err, "failed to write the slim backup manifest")
+	}
+	return errors.Wrap(WriteBackupManifestFilesSST(ctx, exportStore, encryption, kmsEnv, manifest,
+		SlimManifestFileInfoPath), "failed to write backup manfiest files SST")
+}
+
+// WriteSlimBackupManifest compresses and writes a slim version of the
+// BackupManifest `desc` to `exportStore`.
+//
+// A slim backup manifest has the following repeated fields of the
+// BackupManifest proto nil'ed out:
+// - Files
+func writeSlimBackupManifest(
+	ctx context.Context,
+	exportStore cloud.ExternalStorage,
+	filename string,
+	encryption *jobspb.BackupEncryptionOptions,
+	kmsEnv cloud.KMSEnv,
+	manifest *backuppb.BackupManifest,
+) error {
+	slimManifest := *manifest
+	slimManifest.Files = nil
```

Comment:
I just remembered a conversation about writing a sentinel bad file here instead of nil'ing the field out, i'll amend the patch but this should still be rfal!

Reply:
donezo